lib/Backend/IRBuilder.cpp

```diff
@@ -750,28 +751,41 @@ IRBuilder::Build()
             }
         }
 #endif
-        if (IsLoopBodyInTry() && m_lastInstr->GetDst() && m_lastInstr->GetDst()->IsRegOpnd() && m_lastInstr->GetDst()->GetStackSym()->HasByteCodeRegSlot())
+
+        if (IsLoopBodyInTry() && lastProcessedInstrForJITLoopBody != m_lastInstr)
         {
-            StackSym * dstSym = m_lastInstr->GetDst()->GetStackSym();
-            Js::RegSlot dstRegSlot = dstSym->GetByteCodeRegSlot();
-            if (!this->RegIsTemp(dstRegSlot) && !this->RegIsConstant(dstRegSlot))
+            FOREACH_INSTR_BACKWARD_EDITING_IN_RANGE(
```

Comment:
Please add a comment to make this more obvious.

---
In reply to: [100937161](https://github.com/Microsoft/ChakraCore/pull/2518#discussion_r100937161) [](ancestors = 100937161,100936203)